# HSB 21030-10: Internal load distribution of fastener group 
HSB
Issue D Year 1989

Groth Schneider 13.12.1989 MBB-UF


In [1]:
import sys

sys.path.append("..")

#import markdown
# show latex from string
from IPython.display import display, Markdown

from pylantir.taters import get_markdown_from_docs

from pylantir.pyelbe.hsb.hsb_formulas import (
    moment_x_reference,
    moment_y_reference,
    moment_z_reference,
    moment_x_reference_markdown,
    moment_y_reference_markdown,
    moment_z_reference_markdown,
)


## Summary: 
This sheet provides a method to calculate the shear and tensile forces in fastener groups.  
  
  
Key Words: Fastener group, shear force, tension force

# 1. General
This sheet provides a method to calculate the shear and tensile forces for each fastener in riveted or  
bolted joints. The method is valid only for components that are sufficiently stiff in the joint area.  
For the static strength analyses the centroid of the fastener group is determined taking into account  
the allowable forces of the fasteners. For fatigue analyses the stiffnesses of the fasteners should be  
taken into account. The configuration considered is as follows (note: support forces are not shown):  

<img src="../docs/source/_static/HSB_21030_10_fig_1_Overview.PNG" alt="Drawing" style="width: 500px;"/>

# 2. List of Symbols and Indices  

Symbol | Unit | Description  
---|---|---
x/y/z | mm | cartesian coordinates  
F | N | force  
M | Nmm | moment  
RF | - | reserve factor  
α | ◦ | angle of principal axes  
  
  


  
Index | Description
---|---
<sub>all<sub> | allowable
<sub>i<sub> | index for the fastener i
<sub>s/t<sub> | shear / tension
<sub>C/P<sub> | contact point / point where external loads are applied
<sub>S/T<sub> | centroid for the fastener shear / tensile forces
<sub>U<sub> | reference point in the plane of the fastener group


# 3. Analysis
## 3.1. Transformation of loading to the reference point U
<img src="../docs\source\_static\HSB_21030_10_fig_2_reference__point_U_.PNG" alt="shear forces" style="width: 500px;"/>

In [2]:
Markdown(get_markdown_from_docs(moment_x_reference.__doc__))

$$M_{xU} = M_{xP} - F_{y}\cdot (z_{P} - z_{U}) + F_{z}\cdot (y_{P} - y_{U})$$

In [3]:

Markdown(get_markdown_from_docs(moment_y_reference.__doc__))


$$M_{yU} = M_{yP} + F_{x}\cdot (z_{P} - z_{U}) - F_{z}\cdot (x_{P} - x_{U})$$

In [4]:
Markdown(get_markdown_from_docs(moment_z_reference.__doc__))

$$M_{zU} = M_{zP} - F_{x}\cdot (y_{P} - y_{U}) + F_{y}\cdot (x_{P} - x_{U})$$

## 3.2 Shear forces
<img src="../docs\source\_static\HSB_21030_10_fig_3_Shear_forces.PNG" alt="shear forces" style="width: 500px;"/>

### 3.2.1 Centroid of the fastener group (point S)


In [5]:
from pylantir.pyelbe.hsb.hsb_21030_10 import FastenerGroup, Hsb2103001

In [6]:
Markdown(get_markdown_from_docs(FastenerGroup.calculate_centroid_ys.__doc__))

$$y_{S} = \frac{\sum (F_{s,all,i}\cdot y_{i})}{\sum F_{s,all,i}}$$

In [7]:
Markdown(get_markdown_from_docs(FastenerGroup.calculate_centroid_zt.__doc__))

$$z_{T} = \frac{\sum (F_{t,all,i}\cdot z_{i})}{\sum F_{t,all,i}}$$

In [8]:
Markdown(get_markdown_from_docs(Hsb2103001.calculate_mxs.__doc__))

$$M_{xS} = M_{xU} + F_{y} \cdot z_{s} - F_{z} \cdot y_{s}$$

### 3.2.3 Shear forces in the fasteners


In [9]:
Markdown(get_markdown_from_docs(Hsb2103001.calculate_fsy.__doc__))

$$F_{S,y,i} = F_{y} + \frac {F_{S,all,i}}{\sum F_{S,all,i}} -
        M_{xS} \cdot \frac{F_{S,all,i} \cdot (z_{i} - z_{s})}{\sum \left
        \{ F_{S,all,i} \cdot [(y_{i} - y_{S} )^{2} + (z_{i} - z_{s})^{2}] \right \}}$$

In [10]:
Markdown(get_markdown_from_docs(Hsb2103001.calculate_fsz.__doc__))

$$F_{S,z,i} = F_{z} - \frac{F_{S,all,i}}{\sum F_{S,all,i}} +
        M_{x,S} \cdot \frac{F_{S,all,i} \cdot (y_{i} - y_{s})}{\sum \left
        \{ F_{S,all,i} \cdot [(y_{i} - y_{S} )^{2} + (z_{i} - z_{s})^{2}] \right \}}$$

In [11]:
Markdown(get_markdown_from_docs(Hsb2103001.calculate_fastener_shear_forces.__doc__))

$$F_{S,i} = \sqrt{F_{S,z,i}^{2} + F_{S,y,i}^{2}}$$

## 3.3 Tensile forces  

If the fasteners are made of different materials, the allowable tensile forces need to be scaled by the  
ratio of the moduli of elasicity.  
Forces taken by the fasteners:  
<img src="../docs\source\_static\HSB_21030_10_fig_5_Tensile_forces.PNG" alt="shear forces" style="width: 500px;"/>  
### 3.3.1 Centroid of the fastener group (point T)


In [12]:
Markdown(get_markdown_from_docs(FastenerGroup.calculate_centroid_yt.__doc__))

$$y_{T} = \frac{\sum (F_{t,all,i}\cdot y_{i})}{\sum F_{t,all,i}}$$

In [13]:
Markdown(get_markdown_from_docs(FastenerGroup.calculate_centroid_zt.__doc__))

$$z_{T} = \frac{\sum (F_{t,all,i}\cdot z_{i})}{\sum F_{t,all,i}}$$

### 3.3.2 Moments about point T


In [14]:
Markdown(get_markdown_from_docs(Hsb2103001.calculate_mys.__doc__))

$$M_{yS} = M_{yU} - F_{x} \cdot z_{T}$$

In [15]:
Markdown(get_markdown_from_docs(Hsb2103001.calculate_mzs.__doc__))

$$M_{zS} = M_{zU} + F_{x} \cdot y_{T}$$

<img src="../docs\source\_static\HSB_21030_10_fig_4_Moments_about_point_T.PNG" alt="shear forces" style="width: 500px;"/>  

### 3.3.3 Tensile forces in the fasteners  
To balance the forces it is necessary to transform the applied loading into the principal axis system  
of the fastener group.

<img src="../docs\source\_static\HSB_21030_10_fig_6_Tensile_forces_transformation.PNG" alt="shear forces" style="width: 500px;"/>  

In [16]:
Markdown(get_markdown_from_docs(Hsb2103001.calculate_alpha.__doc__))

$$tan(2 \cdot \alpha) = 2 \cdot \frac{\sum[F_{T,all,i} \cdot (y_{i} -
         y_{T}) \cdot (z_{i} - z_{T})]}
        {\sum[F_{T,all,i} \cdot (y_{i} - y_{T})^{2}] - \sum[F_{T,all,i} \cdot (z_{i} - z_{T})^{2}]}$$

In [17]:
Markdown(get_markdown_from_docs(Hsb2103001.calculate_centroid_yta.__doc__))

$$y_{T,a} = y_{T} * cos(\alpha) - z_{T} * sin(\alpha)$$

In [18]:
Markdown(get_markdown_from_docs(Hsb2103001.calculate_centroid_zta.__doc__))

$$z_{T,a} =  - y_{T} * sin(\alpha) + z_{T} * cos(\alpha)$$

In [19]:
Markdown(get_markdown_from_docs(Hsb2103001.calculate_fastener_ya.__doc__))

$$y_{a,i} = y_{i} * cos(\alpha) - z_{i} * sin(\alpha)$$

In [20]:
Markdown(get_markdown_from_docs(Hsb2103001.calculate_fastener_za.__doc__))

$$z_{a,i} =  - y_{i} * sin(\alpha) + z_{i} * cos(\alpha)$$

In [21]:
Markdown(get_markdown_from_docs(Hsb2103001.calculate_moment_ya.__doc__))

$$M_{ySA} = M_{yS} * cos(\alpha) - M_{zS} * sin(\alpha)$$

In [22]:
Markdown(get_markdown_from_docs(Hsb2103001.calculate_moment_za.__doc__))

$$M_{zSA} =  - M_{yS} * sin(\alpha) + M_{zS} * cos(\alpha)$$

In [23]:
Markdown(get_markdown_from_docs(Hsb2103001.calculate_fastener_tension_force_f1.__doc__))

$$F_{1,i} = F_{x}\cdot \frac{F_{t,all,i}}{\sum F_{t,all,i}}$$

In [24]:
Markdown(get_markdown_from_docs(Hsb2103001.calculate_fastener_tension_force_f2.__doc__))

$$F_{2,i} = M_{ySA}\cdot \frac{\left [ F_{T,all,i} \cdot
                \left ( z_{Ai} - z_{TA} \right ) \right ]}{\sum \left [ F_{T,all,i} \cdot
                \left ( z_{Ai} - z_{TA} \right )^2 \right ]}$$

In [25]:
Markdown(get_markdown_from_docs(Hsb2103001.calculate_fastener_tension_force_f3.__doc__))

$$F_{2,i} = M_{zSA}\cdot \frac{\left [ F_{T,all,i} \cdot
                \left ( y_{Ai} - y_{TA} \right ) \right ]}{\sum \left [ F_{T,all,i} \cdot
                \left ( y_{Ai} - y_{TA} \right )^2 \right ]}$$

In [26]:
Markdown(get_markdown_from_docs(Hsb2103001.calculate_fastener_tension_force.__doc__))

$$F_{i} = F_{1,i} + F_{2,i} + F_{3,i}$$

## 3.4 Additional forces due to the contact of the joined parts (compression)  
The fasteners considered above are loaded in tension only. Compressive forces are taken by contact  
of the joined parts (point C). If, after the first calculation, individual fasteners show negative tensile  
forces, a second calculation is required as follows:  

<img src="../docs\source\_static\HSB_21030_10_fig_7_Compression.PNG" alt="shear forces" style="width: 500px;"/>

The allowable tensile force of those fasteners that experience negative tensile forces after the first  
iteration is set to zero (0.001 N). The allowable shear forces remain unchanged.  
In the area of the two faces in contact, introduce a dummy fastener with an allowable shear force  
of zero (0.001 kN) and a large allowable tensile force (9999 kN). The resulting negative tensile  
force of the dummy fastener represents the supporting force FC of the contact. The position of  
the dummy fastener is the point where the resulting compressive force acts; the position has to be  
estimated based on the joint stiffness.  

# 4 Example

## 4.1 Configuration considered
### 4.1.1 Layout of the fastener group

<img src="../docs\source\_static\HSB_21030_10_fig_8_Layout_of_the_fastener_group.PNG" alt="shear forces" style="width: 500px;"/>

### 4.1.2 Contact area of the joint parts (no load transfer)

<img src="../docs\source\_static\HSB_21030_10_fig_9_Contact_area_of_the_joint_parts.PNG" alt="shear forces" style="width: 500px;"/>

### 4.1.3 Contact area and reaction forces (kN)

<img src="../docs\source\_static\HSB_21030_10_fig_10_Contact_area_and_reaction_forces.PNG" alt="shear forces" style="width: 500px;"/>

## 4.2 Loading of the fastener group (ultimate, with respect to point P)

xP = 30.0 mm | yP = 0 mm | zP = 0 mm  
---|---|---  
Fx = +10.0 kN | Fy = +12.0 kN | Fz = −2.0 kN  
Mx = −240 N · m M | My = +200 N · m | Mz = 0  





In [27]:
from pylantir.pyelbe.hsb.hsb_21030_10 import ReferencePoint, Forces, Moments, Fastener, Hsb2103001


# point P definition
point_p = ReferencePoint(name="P", x_coord=30, y_coord=0, z_coord=0)

# point U definition
point_u = ReferencePoint(name="P", x_coord=0, y_coord=0, z_coord=0)

# Load definition
# force around P
# Forces = namedtuple("Forces", "force_x force_y force_z")
forces = Forces(name="forces", force_x=10000, force_y=12000, force_z=-2000)

# Moment around U
moments = Moments(name="moments", moment_x=-240000, moment_y=200000, moment_z=0)

    # fastener 1 definition
    # fastener_1 = Fastener("fast1", "test", "test", 1, 1, 1, 1, 0, -70, 35)
fastener_1 = Fastener(
    name="fast1",
    specification="test",
    material="test",
    shear_allowable=18500,
    tension_allowable=12000,
    x_coord=0,
    y_coord=-70,
    z_coord=35,
)

# fastener 2 definition
# fastener_2 = Fastener("fast2", "test", "test", 1, 1, 1, 1, 0, -40, 35)
fastener_2 = Fastener(
    name="fast2",
    specification="test",
    material="test",
    shear_allowable=18500,
    tension_allowable=12000,
    x_coord=0,
    y_coord=-40,
    z_coord=35,
)

# fastener 3 definition
# fastener_3 = Fastener("fast3", "test", "test", 1, 1, 1, 1, 0, -40, 15)
fastener_3 = Fastener(
    name="fast3",
    specification="test",
    material="test",
    shear_allowable=18500,
    tension_allowable=12000,
    x_coord=0,
    y_coord=-40,
    z_coord=15,
)

# fastener 4 definition
# fastener_4 = Fastener("fast4", "test", "test", 1, 1, 1, 1, 0, -60, 15)
fastener_4 = Fastener(
    name="fast4",
    specification="test",
    material="test",
    shear_allowable=18500,
    tension_allowable=12000,
    x_coord=0,
    y_coord=-60,
    z_coord=15,
)

# fastener group definition
fastener_group = FastenerGroup(
    name="test", fasteners=[fastener_1, fastener_2, fastener_3, fastener_4]
)

# Hsb2103001
hsb_calc = Hsb2103001(
    name="Hsb2103001",
    fastener_group=fastener_group,
    forces=forces,
    moments=moments,
    application_point=point_p,
    reference_point=point_u,
)


fastener fast1 is in compression: -1115.3846153846134
fastener fast2 is in tension: 6615.384615384614
fastener fast3 is in tension: 4826.923076923076
fastener fast4 is in compression: -326.9230769230744
Fasteners in compression, iterate calculation!


## 4.3 Moments about point U
MxU = −240 N · m MyU = +260 N · m MzU = +360 N · m

In [28]:
print(hsb_calc.moments_u) 

moments_u(moment_x_u=-240000.0, moment_y_u=260000.0, moment_z_u=360000.0)


## 4.4 Centroid of the fastener group (with respect to point U)
Shear 
yS = −52.5 mm 
zS = +25.0 mm  

Tension 
yT = −52.5 mm 
zT = +25.0 mm  

In [29]:
print(f"{hsb_calc.fastener_group.centroid_ys = }")
print(f"{hsb_calc.fastener_group.centroid_zs = }")
print()
print(f"{hsb_calc.fastener_group.centroid_yt = }")
print(f"{hsb_calc.fastener_group.centroid_zt = }")


hsb_calc.fastener_group.centroid_ys = -52.5
hsb_calc.fastener_group.centroid_zs = 25.0

hsb_calc.fastener_group.centroid_yt = -52.5
hsb_calc.fastener_group.centroid_zt = 25.0


## 4.5 Moments about the centroid of the fastener group
Shear MxS = −45 N · m  
Tension MyS = +10 N · m  
Tension MzS = −165 N · m  

In [30]:
print(f"{hsb_calc.moment_x_s = }")
print(f"{hsb_calc.moment_y_s = }")
print(f"{hsb_calc.moment_z_s = }")
print("NOTE: units are in Nmm")

hsb_calc.moment_x_s = -45000.0
hsb_calc.moment_y_s = 10000.0
hsb_calc.moment_z_s = -165000.0
NOTE: units are in Nmm


# 4.6 Forces of the individual fasteners
yi | zi |  Fs,y,i |  Fs,z,i |  Fs,i |  Ft,i
---|---|---|---|---|---
i |  mm |  mm |  kN |  kN |  kN |  kN
1 |  -70.0 |  +35.0 |  +3.42 |  +0.23 |  3.43 |  -1.12
2 |  -40.0 |  +35.0 |  +3.42 |  -1.02 |  3.57 |  +6.62
3 |  -40.0 |  +15.0 |  +2.58 |  -1.02 |  2.78 |  +4.83
4 |  -60.0 |  +15.0 |  +2.58 |  -0.19 |  2.59 |  -0.33

In [31]:
import pandas as pd
pd.DataFrame(hsb_calc.result_dict)

X   Y  Shear  Tension         Fsyi          Fsz  Shear Force  \
0  0 -70  18500    12000  3418.604651   232.558140  3426.505662   
1  0 -40  18500    12000  3418.604651 -1023.255814  3568.460483   
2  0 -40  18500    12000  2581.395349 -1023.255814  2776.806512   
3  0 -60  18500    12000  2581.395349  -186.046512  2588.091044   

   Tension Force  Shear Reserve Factor  Tension Reserve Factor  
0   -1115.384615                  5.39                  -10.75  
1    6615.384615                  5.18                    1.81  
2    4826.923077                  6.66                    2.48  
3    -326.923077                  7.14                  -36.70

## 4.7 Fasteners loaded in compression
Fasteners 1 and 4 are loaded in compression (Ft,i is negative). The determination of the fastener  
forces is repeated with Ft,all for the fasteners 1 and 4 set zero, both fasteners are therefore loaded  
in shear only. A dummy fastener is introduced. This fastener (5) represents the location where the  
resulting compressive force acts: location C.  

In [32]:
from pylantir.pyelbe.hsb.hsb_21030_10 import iterate_calc

hsb_calc_iteration = iterate_calc(hsb_calc)


fastener fast1 is in tension: 0.0
fastener fast2 is in tension: 6166.666666666668
fastener fast3 is in tension: 5166.666666666668
fastener fast4 is in tension: 0.0
fastener dummy_fastener is in compression: -11189.089874089555
Fasteners in compression, iterate calculation!
NOTE: fastener dummy_fastener is added to fastener group


### 4.7.1 Centroid of the fastener group (with respect to point P) (iteration of 4.4)
Shear  
yS = −52.5 mm   
zS = +25.0 mm  
  
Tension   
yT = −69.9 mm   
zT = +25.0 mm  

In [33]:
print(f"{hsb_calc_iteration.fastener_group.centroid_ys = }")
print(f"{hsb_calc_iteration.fastener_group.centroid_zs = }")
print()
print(f"{hsb_calc_iteration.fastener_group.centroid_yt = }")
print(f"{hsb_calc_iteration.fastener_group.centroid_zt = }")


hsb_calc_iteration.fastener_group.centroid_ys = -52.500000023648646
hsb_calc_iteration.fastener_group.centroid_zs = 24.999999999999996

hsb_calc_iteration.fastener_group.centroid_yt = -69.92816521999401
hsb_calc_iteration.fastener_group.centroid_zt = 25.0


## 4.7.2 Moments about the centroid of the fastener group (iteration of 4.5)
Shear Mx,S = −45 N · m  
Tension My,S = +10 N · m  
Tension Mz,S = −339 N · m  

In [34]:
print(f"{hsb_calc_iteration.moment_x_s = }")
print(f"{hsb_calc_iteration.moment_y_s = }")
print(f"{hsb_calc_iteration.moment_z_s = }")
print("NOTE: units are in Nmm")

hsb_calc_iteration.moment_x_s = -45000.00004729735
hsb_calc_iteration.moment_y_s = 10000.0
hsb_calc_iteration.moment_z_s = -339281.6521999402
NOTE: units are in Nmm


### 4.7.3 Forces of the individual fastener (iteration of 4.6)
fastener  | yi | zi |  Fs,y,i |  Fs,z,i |  Fs,i |  Ft,i
---|---|---|---|---|---|---
i |  mm |  mm |  kN |  kN |  kN |  kN |  kN
1 |  -70.0 |  +35.0 |  +3.42 |  +0.23 |  3.43 |  0
2 |  -40.0 |  +35.0 |  +3.42 |  -1.02 |  3.57 |  +6.17
3 |  -40.0 |  +15.0 |  +2.58 |  -1.02 |  2.78 |  +5.17
4 |  -60.0 |  +15.0 |  +2.58 |  -0.19 |  2.59 |  0
“5” |-70.0 |  +25.0 |  0 |  0 |  0 |  -1.33

### 4.7.4 Reserve Factors of the individual fasteners
For the fastener type HL10 (Hi-Lok, diameter = 6 mm with Fs,all,i and Ft,all,i taken from the corresponding HSB sheets) and with Fs,i and Ft,i taken from the table given above the reserve factors
RF follow as:

Fastener | Fs,all,i |  Ft,all,i |  RF = Fs,all,i/Fs,i |  RF = Ft,all,i/Ft,i
---|---|---|---|---
i |  kN |  kN |  - |  -
1 |  18.5 |  -  | 18.5/3.43 = 5.39  | -
2 |  18.5 |  12.0 |  18.5/3.57 = 5.18 |  12.0/6.17 = 1.92
3 |  18.5 | 12.0  | 18.5/2.78 = 6.65  | 12.0/5.17 = 2.27
4 |  18.5 | -  | 18.5/2.59 = 7.14 |  -

In [35]:
pd.DataFrame(hsb_calc_iteration.result_dict)

X   Y       Shear  Tension         Fsyi          Fsz  Shear Force  \
0  0.0 -70  18500.0000        0  3418.604647   232.558139  3426.505657   
1  0.0 -40  18500.0000    12000  3418.604647 -1023.255814  3568.460479   
2  0.0 -40  18500.0000    12000  2581.395345 -1023.255814  2776.806509   
3  0.0 -60  18500.0000        0  2581.395345  -186.046512  2588.091040   
4  0.0 -70      0.0001  9999000     0.000016     0.000001     0.000016   

   Tension Force  Shear Reserve Factor  Tension Reserve Factor  
0       0.000000                  5.39                     NaN  
1    6166.666667                  5.18                    1.94  
2    5166.666667                  6.66                    2.32  
3       0.000000                  7.14                     NaN  
4  -11189.089874                  6.14                 -893.63